File: src/main/java/org/taktik/icure/services/external/rest/v1/dto/embed/PatientHealthCarePartyDto.java
/*
 * Copyright (C) 2018 Taktik SA
 *
 * This file is part of iCureBackend.
 *
 * iCureBackend is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License version 2 as published by
 * the Free Software Foundation.
 *
 * iCureBackend is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with iCureBackend.  If not, see <http://www.gnu.org/licenses/>.
 */

package org.taktik.icure.services.external.rest.v1.dto.embed;

/** Created by aduchate on 02/07/13, 11:59 */


import java.io.Serializable;
import java.util.List;
import java.util.Map;

public class PatientHealthCarePartyDto implements Serializable {
	private static final long serialVersionUID = 1L;

    protected PatientHealthCarePartyTypeDto type;
    @Deprecated
    protected boolean referral = false; // true if healthcarePartyId is DMG current holder/owner
    protected String healthcarePartyId;
    protected Map<TelecomType,String> sendFormats;  // String is in fact a UTI (uniform type identifier / a sort of super-MIME)
    protected List<ReferralPeriod> referralPeriods; // History of DMG ownerships

    public String getHealthcarePartyId() {
        return healthcarePartyId;
    }

    public void setHealthcarePartyId(String healthcarePartyId) {
        this.healthcarePartyId = healthcarePartyId;
    }

    public Map<TelecomType, String> getSendFormats() {
        return sendFormats;
    }

    public void setSendFormats(Map<TelecomType, String> sendFormats) {
        this.sendFormats = sendFormats;
    }

	public boolean isReferral() {
		return referral;
	}

	public void setReferral(boolean referral) {
		this.referral = referral;
	}

    public List<ReferralPeriod> getReferralPeriods() {
        return referralPeriods;
    }

    public void setReferralPeriods(List<ReferralPeriod> referralPeriods) {
        this.referralPeriods = referralPeriods;
    }

	public PatientHealthCarePartyTypeDto getType() {
		return type;
	}

	public void setType(PatientHealthCarePartyTypeDto type) {
		this.type = type;
	}


}

File: src/main/kotlin/org/taktik/icure/services/external/rest/v1/dto/embed/PatientHealthCarePartyDto.kt
/*
 *  iCure Data Stack. Copyright (c) 2020 Taktik SA
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU Affero General Public License as
 *     published by the Free Software Foundation, either version 3 of the
 *     License, or (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful, but
 *     WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *     Affero General Public License for more details.
 *
 *     You should have received a copy of the GNU Affero General Public
 *     License along with this program.  If not, see
 *     <https://www.gnu.org/licenses/>.
 */
package org.taktik.icure.services.external.rest.v1.dto.embed

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import com.github.pozo.KotlinBuilder
import io.swagger.v3.oas.annotations.media.Schema
import java.io.Serializable
import java.util.*

/**
 * Created by aduchate on 02/07/13, 11:59
 */

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
@KotlinBuilder
@Schema(description = "One or several periods of care by an hcp for this patient")
data class PatientHealthCarePartyDto(
        @Schema(description = "Type of care/relationship.") val type: PatientHealthCarePartyTypeDto? = null,
        @Schema(description = "UUID of the hcp.") val healthcarePartyId: String? = null,
        @Schema(description = "Preferred format of exchange for diverse means of communication") val sendFormats: Map<TelecomTypeDto, String> = mapOf(),  // String is in fact a UTI (uniform type identifier / a sort of super-MIME)
        @Schema(description = "Time periods") val referralPeriods: SortedSet<ReferralPeriodDto> = sortedSetOf(), // History of DMG ownerships
        @get:Deprecated("Use type") @Schema(defaultValue = "false") val referral: Boolean = false, // mark this phcp as THE active referral link (gmd)
        override val encryptedSelf: String? = null
) : EncryptedDto, Serializable
